core-api/src/main/java/com/optimizely/ab/internal/ExperimentUtils.java

```diff
@@ -59,22 +63,53 @@ public static boolean isExperimentActive(@Nonnull Experiment experiment) {
     public static boolean isUserInExperiment(@Nonnull ProjectConfig projectConfig,
                                              @Nonnull Experiment experiment,
                                              @Nonnull Map<String, ?> attributes) {
+        if (experiment.getAudienceConditions() != null) {
+            Boolean resolveReturn = resolveAudienceConditions(projectConfig, experiment, attributes);
+            return resolveReturn == null ? false : resolveReturn;
+        }
+        else {
+            Boolean resolveReturn = resolveAudience(projectConfig, experiment, attributes);
+            return resolveReturn == null ? false : resolveReturn;
+        }
+    }
+
+    public static @Nullable Boolean resolveAudience(@Nonnull ProjectConfig projectConfig,
```

Comment:
Maybe rename to `resolveAudienceIds`, for symmetry with `resolveAudienceConditions`? (Or `evaluateAudienceIds` and `evaluateAudienceConditions`, though I don't really care)